examples/derived_gauges/README.md

```diff
@@ -0,0 +1,348 @@
+# Derived Gauge Example
+
+Table of Contents
+=================
+- [Summary](#summary)
+- [Run the example.](#run-the-example)
+- [How to use derived gauges?](#how-to-use-derived-gauges-)
+  * [Initialize Metric Registry](#initialize-metric-registry)
+  * [Create derived gauge metric](#create-derived-gauge-metric)
+  * [Create derived gauge entry](#create-derived-gauge-entry)
+  * [Implement derived gauge interface](#implement-derived-gauge-interface)
+  * [Complete Example](#complete-example)
+
+
+
+## Summary 
+[top](#Table-of-Contents)
+
+This example demonstrates the use of derived gauges. It is a simple interactive program of consumer
+and producer. User can input number of items to produce. Producer produces specified number of
+items. Consumer consumes randomly consumes 1-5 items in each attempt. It then sleeps randomly
+between 1-10 seconds before the next attempt.
+
+There are two metrics collected to monitor the queue.
+1. **queue_size**: It is an instantaneous queue size represented using derived gauge int64.
+1. **queue_seconds_since_processed_last**: It is the time elaspsed in seconds since the last time
+   when the queue was consumed. It is represented using derived gauge float64.
+This example shows how to use gauge metrics. The program records two gauges.
+
+These metrics are read when exporter scrapes them. In these example prometheus exporter is used to
+scrape the data. Metrics can be viewed at [http://localhost:9090/metrcs](http://localhost:9090/metrcs) once the program is running.
+
+Enter different value for number of items to queue and fetch the metrics using above url to see the variation in the metrics.
+
+## Run the example.
+
+```
+$ go get go.opencensus.io/examples/derived_gauges/...
+```
+
+then:
+
+```
+$ go run $(go env GOPATH)/src/go.opencensus.io/examples/derived_gauges/derived_gauge.go
+```
+
+## How to use derived gauges?
+
+### Initialize Metric Registry
+Create a new metric registry for all your metrics.
+This step is a general step for any kind of metrics and not specific to gauges.
+Register newly created registry with global producer manager.
+
+[embedmd]:# (derived_gauge.go reg)
+```go
+r := metric.NewRegistry()
+metricproducer.GlobalManager().AddProducer(r)
+```
+
+
+### Create derived gauge metric
+Create a gauge metric. In this example we have two metrics.
+
+**queue_size**
+
+[embedmd]:# (derived_gauge.go size)
+```go
+queueSizeGauge, err := r.AddInt64DerivedGauge(
+	"queue_size",
+	metric.WithDescription("Instantaneous queue size"),
+	metric.WithUnit(metricdata.UnitDimensionless))
+if err != nil {
+	log.Fatalf("error creating queue size derived gauge, error%v\n", err)
```

Comment:
Nit: add a space between "error" and "%v"

Reply:
done.